pkg/apis/keepalive/v1beta1/register.go

```diff
@@ -0,0 +1,74 @@
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/intstr"
+)
+
+// VirtualIP is a keepalived CRD specificiation for virtual IP addresses
+type VirtualIP struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata"`
+
+	Spec   VirtualIPSpec `json:"spec"`
+	Status Status        `json:"status"`
+}
+
+// VirtualIPSpec defines the spec of the CRD
+type VirtualIPSpec struct {
+	// IP virtual IP to use
+	IP string `json:"ip"`
+	// Port of the virtual
+	Port int `json:"port"`
+
+	// Interface interface where the virtual IP should be configured
+	// The default value is eth0
+	Interface *string `json:"interface,omitempty"`
+
+	// ProxyProtocol indicates if proxy-protocol is enabled in the service being exposed
+	ProxyProtocol bool `json:"proxyProtocol"`
+	//Protocol defines the protocol of the service. Valid values are TCP and UDP
+	Protocol corev1.Protocol `json:"protocol"`
+	// ServiceReference defines a reference to the service to expose
+	ServiceReference ServiceReference `json:"serviceReference"`
```

Comment:
Like? Do you have an example?

Reply:
I mean how will this CRD support the current functionality that you can leave the service as empty string and it will create the VIP as below:
```yaml
apiVersion: v1
data:
  10.0.22.170: ""
kind: ConfigMap
metadata:
  name: keepalived
```

Reply:
An empty [[]ServiceReference](https://github.com/aledbf/kube-keepalived-vip/pull/85/files#diff-223de3097c318a65d59649cc09f01d5cR31) definition?

Reply:
Sounds good, just wanted to make sure it would be supported.